cmake/config.cmake

```diff
@@ -198,6 +198,16 @@ set(USE_DNNL_CODEGEN OFF)
 set(USE_ARM_COMPUTE_LIB OFF)
 set(USE_ARM_COMPUTE_LIB_GRAPH_RUNTIME OFF)
 
+# Whether to build with Arm Ethos-N support
+# Possible values:
+# - OFF: disable Arm Ethos-N support
+# - path/to/arm-ethos-N-stack: use a specific version of the
+#   Ethos-N driver stack
+set(USE_ETHOSN OFF)
+# If USE_ETHOSN is enabled, use Ethos-N hardware (ON) or
+# software test infrastructure (OFF)
+set(USE_ETHOSN_HW ON)
```

Comment:
Good catch, yes it should be off by default. The naming of USE_ETHOSN is really because it toggles both the codegen and runtime support. USE_ETHOSN_HW determines whether or not we mock the inference function in the runtime or use real hardware.

Reply:
Ah I see. Then I'd suggest the comments for this flag to something like:
```
# Whether Ethos-N is available on this machine. Software test infra will be used when OFF.
set(USE_ETHOSN_HW OFF)
```

And set up a checker for the case that USE_ETHOSN_HW=ON but USE_ETHOSN=OFF.